apps/worker/src/blockchain/extrinsic.ts

```diff
@@ -63,54 +62,24 @@ export class Extrinsic<T extends ISubmittableResult = ISubmittableResult, C exte
     return this.event;
   }
 
-  public signAndSend(nonce?: number): Promise<ParsedEventResult> {
-    return firstValueFrom(
-      this.extrinsic.signAndSend(this.keys, { nonce }).pipe(
-        filter(({ status }) => status.isInBlock || status.isFinalized),
-        this.parseResult(this.event),
-      ),
+  public signAndSend(nonce?: number): Promise<[Hash, EventMap]> {
+    return firstValueFrom(this.extrinsic.signAndSend(this.keys, { nonce }).pipe(filter(({ status }) => status.isInBlock || status.isFinalized))).then(
+      ({ status, events, txHash }) => {
+        if (status.isInBlock || status.isFinalized) {
+          const eventMap: EventMap = {};
+          events.forEach((record: EventRecord) => {
+            const { event } = record;
+            eventMap[eventKey(event)] = event;
+          });
+          return [txHash, eventMap];
```

Comment:
simply return txHash and eventMap, dont wait to `parseResult` will be done by status monitoring processor?